src/poetry/core/packages/vcs_dependency.py

```diff
@@ -176,3 +176,12 @@ def __str__(self) -> str:
 
     def __hash__(self) -> int:
         return hash((self._name, self._vcs, self._branch, self._tag, self._rev))
+
+    def __eq__(self, other: "VCSDependency") -> int:
```

Comment:
Note: I'm not affiliated with this project in any capacity, just a bystander, but mypy checks failed with the following error text: 

```
src/poetry/core/packages/vcs_dependency.py:180: error: Return type "int" of "__eq__" incompatible with return type "bool" in supertype "Dependency"
src/poetry/core/packages/vcs_dependency.py:180: error: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the argument type as "object"
src/poetry/core/packages/vcs_dependency.py:180: note: This violates the Liskov substitution principle
src/poetry/core/packages/vcs_dependency.py:180: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
src/poetry/core/packages/vcs_dependency.py:180: note: It is recommended for "__eq__" to work with arbitrary objects, for example:
src/poetry/core/packages/vcs_dependency.py:180: note:     def __eq__(self, other: object) -> bool:
src/poetry/core/packages/vcs_dependency.py:180: note:         if not isinstance(other, VCSDependency):
src/poetry/core/packages/vcs_dependency.py:180: note:             return NotImplemented
src/poetry/core/packages/vcs_dependency.py:180: note:         return <logic to compare two VCSDependency instances>
src/poetry/core/packages/vcs_dependency.py:180: error: Return type "int" of "__eq__" incompatible with return type "bool" in supertype "object"
```

I would recommend trying a `bool` for the return type of your newly-defined `__eq__`.